### ISSUE TITLE
stack-1.7.1: VersionIntervals.hs:57:3: error: Couldn't match expected type `C.VersionIntervals'

### ISSUE THREAD

**Issue by juhp** [commercialhaskell/stack#3995]
Building with ghc-8.2.2

```
src/Stack/Types/VersionIntervals.hs:57:3: error:
    * Couldn't match expected type `C.VersionIntervals'
                  with actual type `Maybe C.VersionIntervals'
    * In the expression: C.mkVersionIntervals $ map go vi
      In an equation for `toCabal':
          toCabal (VersionIntervals vi)
            = C.mkVersionIntervals $ map go vi
            where
                go (VersionInterval lowerV lowerB mupper)
                  = (C.LowerBound (toCabalVersion lowerV) (toCabalBound lowerB), 
                     case mupper of
                       Nothing -> C.NoUpperBound
                       Just (v, b) -> C.UpperBound (toCabalVersion v) (toCabalBound b))
   |
57 |   C.mkVersionIntervals $ map go vi
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by borsboom** [comment]
How are you building Stack?  All the bindists released for 1.7.1 were built with GHC 8.2.2, so it definitely works.  It seems like your dependency versions may be wrong (not matching what's specified in `stack.yaml`).

**Comment by mpilgrem** [comment]
I am going to close this issue given the passage of time and that Stack no longer supports GHC < 8.4.
